### PR TITLE
fix(components): [table] fix sort-change event behavior

### LINKED ISSUE #4589: [Bug Report] table sortable bug
### Element Plus version
1.2.0-beta.4

### OS/Browsers version
Chrome

### Vue version
3.2.23

### Reproduction Link
https://codepen.io/code-farmer-i/pen/NWaWBNE

### Steps to reproduce
点击三下排序按钮

### What is Expected?
sort-change 回调中 sortable.column  sortable.prop 不为 null

### What is actually happening?
sort-change 回调中 sortable.column sortable.prop 为 null

### LINKED ISSUE #8305: [Feature Request] [Enhancement] table 取消排序没有返回列的prop
Feature type: **`Enhancement`**

## Details

- Enhanced compone nt name: `All`

## Description

table 组件设置了sort-orders：['ascending', 'descending', 'null']后，
在点击箭头进行取消时，返回的column prop为null；
点击列头，返回数据正常；
希望：点击箭头取消时，返回正确的column prop

点击search icon会触发sort事件，希望可以有个参数控制不用点击列的任意位置就触发排序

### Additional comments

_(empty)_

### LINKED ISSUE #8576: [Component] [table] Table header's sort conditon gets lost when listening to sort-change events
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.7`
- Browser / OS: `Chrome/99.0.4844.74 MacOS 12.3`
- Build Tool: `Vite`

### Related Component

- `el-table`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtdGFibGVcbiAgICA6ZGF0YT1cInRhYmxlRGF0YVwiXG4gICAgc3R5bGU9XCJ3aWR0aDogMTAwJVwiXG4gICAgQHNvcnQtY2hhbmdlPVwiaGFuZGxlU29ydENoYW5nZVwiXG4gID5cbiAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJkYXRlXCIgbGFiZWw9XCJEYXRlXCIgc29ydGFibGU9XCJjdXN0b21cIiB3aWR0aD1cIjE4MFwiIC8+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwibmFtZVwiIGxhYmVsPVwiTmFtZVwiIHdpZHRoPVwiMTgwXCIgLz5cbiAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJhZGRyZXNzXCIgbGFiZWw9XCJBZGRyZXNzXCIgLz5cbiAgPC9lbC10YWJsZT5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgdHlwZSB7IFRhYmxlQ29sdW1uQ3R4IH0gZnJvbSAnZWxlbWVudC1wbHVzL2VzL2NvbXBvbmVudHMvdGFibGUvc3JjL3RhYmxlLWNvbHVtbi9kZWZhdWx0cydcblxuaW50ZXJmYWNlIFVzZXIge1xuICBkYXRlOiBzdHJpbmdcbiAgbmFtZTogc3RyaW5nXG4gIGFkZHJlc3M6IHN0cmluZ1xufVxuXG5jb25zdCBoYW5kbGVTb3J0Q2hhbmdlID0gKHBhcmFtczoge1xuICBjb2x1bW46IFRhYmxlQ29sdW1uQ3R4PFVzZXI+IHwgbnVsbFxuICBwcm9wOiBzdHJpbmcgfCBudWxsXG4gIG9yZGVyOiBzdHJpbmcgfCBudWxsXG59KSA9PiB7XG4gIGNvbnNvbGUubG9nKHBhcmFtcylcbn1cblxuY29uc3QgdGFibGVEYXRhOiBVc2VyW10gPSBbXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wMycsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wMicsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNCcsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wMScsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG5dXG48L3NjcmlwdD5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

### Steps to reproduce

1. Click on any sortable header once (Date column for example), to set column order to ascending.
2. Click on the same header for the second time to set sorting condition to descending.
3. Click on the same header once more to unset the sorting condition.

### What is Expected?

On the method handleSortChanged that listens to "sort-change" event.
After the third interaction, I would like to have null value only in the "order" argument to keep a track on my sort object.

For me it makes more sense on click 3 the incoming values on the handler:

Click 3: column: {...}, prop: 'name', order: null

Because, this way, I able to know that a sorting condition (order) on a specific column (prop) has been set to null and then apply it on my remote request.

### What is actually happening?

When the header is clicked the handler to "sort-change" will alternate in the folowing values:

Click 1: column: {...}, prop: 'name', order: 'ascending'
Click 2: column: {...}, prop: 'name', order: 'descending'
Click 3: column: null, prop: null, order: null

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by cokemine**
close https://github.com/element-plus/element-plus/issues/4589

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by MidTin** [comment]
请问这个什么时候能修复？

**Comment by cokemine** [comment]
@sxzz

**Comment by muuyao** [comment]
这个 bug 这么久了不修复么？应该没啥争议吧

**Comment by cokemine** [comment]
hey, @JeremyWuuuuu please take a look

**Comment by wangping1511** [comment]
@sxzz @kooriookami 🙏🏼🙏🏼🙏🏼

File: packages/components/table/src/store/index.ts [hunk 1]
```diff
@@ -142,18 +142,21 @@ function useStore<T>() {
     changeSortCondition(states: StoreStates, options: Sort) {
       // 修复 pr https://github.com/ElemeFE/element/pull/15012 导致的 bug
       const { sortingColumn: column, sortProp: prop, sortOrder: order } = states
-      if (unref(order) === null) {
+      const columnValue = unref(column),
+        propValue = unref(prop),
+        orderValue = unref(order)
```
Comment (tolking) [inline packages/components/table/src/store/index.ts:147]:
```suggestion
     const { sortingColumn, sortProp, sortOrder } = states
      const columnValue = unref(sortingColumn),
        propValue = unref(sortProp),
        orderValue = unref(sortOrder)
```

**Review by tolking — COMMENTED** [review]
抱歉，过了这么久才处理你的PR。我在处理同样问题的PR时，搜索到这个PR，我觉得这是解决问题的正确方式。

我留下些评论供你参考。另外，你愿意增加一个测试吗？

**Review by tolking — APPROVED** [review]
LGTM.

**Pull request merged**